examples/gluon_alexnet/README.md

```diff
@@ -0,0 +1,370 @@
+
+### Load and serve a pre-trained Gluon model
+
+Loading and serving a pre-trained Gluon model is the simplest of the three scenarios. These models don't require you to provide `symbols` and `params` files.
+
+While it is easy to access a model with a couple of lines of code, with MMS you will want to use a [MMS custom service](https://github.com/awslabs/mxnet-model-server/blob/master/docs/custom_service.md) code pattern as follows:
+
+```python
+class MMSPretrainedAlexnet(MXNetVisionService):
+    """
+    Pretrained alexnet Service
+    """
+    def __init__(self, model_name, model_dir, manifest, gpu=None):
+        super(MMSPretrainedAlexnet, self).__init__(model_name, model_dir, manifest, gpu)
+
+        self.net = mxnet.gluon.model_zoo.vision.alexnet(pretrained=True)
+
+    def _preprocess(self, data):
+        img_list = []
+        for idx, img in enumerate(data):
+            input_shape = self.signature['inputs'][idx]['data_shape']
+            # We are assuming input shape is NCHW
+            [h, w] = input_shape[2:]
+            img_arr = mxnet.img.imdecode(img)
+            img_arr = mxnet.image.imresize(img_arr, w, h)
+            img_arr = img_arr.astype(np.float32)
+            img_arr /= 255
+            img_arr = mxnet.image.color_normalize(img_arr,
+                                                  mean=mxnet.nd.array([0.485, 0.456, 0.406]),
+                                                  std=mxnet.nd.array([0.229, 0.224, 0.225]))
+            img_arr = mxnet.nd.transpose(img_arr, (2, 0, 1))
+            img_arr = img_arr.expand_dims(axis=0)
+            img_list.append(img_arr)
+        return img_list
+
+    def _inference(self, data):
+        # Call forward/hybrid_forward
+        output = self.net(data[0])
+        return output.softmax()
+
+    def _postprocess(self, data):
+        idx = data.topk(k=5)[0]
+        return [{'class': (self.labels[int(i.asscalar())]).split()[1], 'probability': float(data[0, int(i.asscalar())].asscalar())}
+                for i in idx]
+```
+
+For an actual code implementation, refer to the custom-service code which uses the [pre-trained Alexnet](https://github.com/awslabs/mxnet-model-server/examples/gluon_alexnet/gluon_pretrained_alexnet.py)
+
+### Serve pre-trained model with MMS
+To serve pre-trained models with MMS we would need to create an model archive file. Follow the below steps to get the example custom service
+loaded and served with MMS.
+
+1. Create a `models` directory
+```bash
+mkdir /tmp/models
+```
+2. Copy the [example code](https://github.com/awslabs/mxnet-model-server/examples/gluon_alexnet/gluon_pretrained_alexnet.py) 
+and other required artifacts to this folder
+```bash
+cp gluon_pretrained_alexnet.py synset.txt signature.json /tmp/models
+```
+3. Run the model-export tool on this folder.
+```bash
+mxnet-model-export --model-name="pretrained-alexnet" --model-path="/tmp/models" --service-file-path="/tmp/models/gluon_pretrained_alexnet.py" --model-type="imperative"
```

Comment:
why do we need another export flag?
to keep it simple, can we infer that from the provided custom code?
maybe if we define a base class (GluonImperativeService) we can infer that from the base class?